Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Plague Scheme

# Changelog:
- The Lich King can start the Plague Scheme:
  - The Plague is limited to 30 counties.
  - Only humans are vulnerable to the Plague.
  - The Plague spreads to counties with high Development and prefers historical Plague regions (so it avoids Hearthglen).
  - Agents accept the Lich Ling's faith and culture after the Plague Scheme.
  - Characters become undead staying in plagued courts.
  - Plagued lands are taken with the Great Invasion CB.
  - Great Invasion war is declared when the Plague starts.
  - Characters that have Mord'truari virtues and the Cynical trait joins the Plague scheme.
  - Plagued counties give special troops based on Development.
- The Lich King trait gives siege progress to Skirmishers and Giants so they can siege too.
- Ner'zhul may declare war on Lordaeron and Quel'thalas without bordering it.
- Ner'zhul has 0 Dark Frenzy in the year 605 bookmark.
- Kel'Thuzard is cynical, ambitious, and deceitful.
- Alexei Barov is greedy, callous, and arrogant.
- Illucia is callous, arrogant, and chaste.
- Pharamond Rivendare is cynical, calm, and callous.

# Chains:
- [x] Ongoing Plague events (happens during the scheming):
  - [x] Event: You spread Death cults to help your scheme. You may choose your strategy.
    - Option 1: Trick cultists into serving, recruit cultists by intrigue
    - Option 2: Sway cultists, recruit cultists by diplomacy.
    - [x] Need text for this, 600-400 characters, from Ner'zhul perspective, in 2nd person
- [x] Plague Outbreak chain (what happens when the scheme is finished):
  - [x] Event: You orders (or may refuse) cultists to infect grain and wood sources
     - [x] Need text for this, 600-400 characters, from Ner'zhul perspective, in 2nd person
  - [x] Event: You succeed, the Plague spreads
     - [x] Need text for this, 600-400 characters, from Ner'zhul perspective, in 2nd person
     - [x] The last 150 characters should tell if the cultists were captured and told who they serve
  - [x] Event: You failed, the Plague prevented
     - [x] Need text for this, 600-400 characters, from Ner'zhul perspective, in 2nd person
     - [x] The last 150 characters should tell if the cultists were captured and told who they serve
  - [x] Event: Your lands are plagued, a notification
     - [x] Need text for this, 600-400 characters, from the victim ruler perspective, in 2nd person
     - [x] The last 150 characters should tell if the cultists were captured and told who they serve
  - [x] Event: Your guards prevented the Plague, a notification
     - [x] Need text for this, 600-400 characters, from the victim ruler perspective, in 2nd person
     - [x] The last 150 characters should tell if the cultists were captured and told who they serve
- [x] Character got the Plague started by the Plaguer and turned undead
  - [x] Need text for this, 600-400 characters, from the sick person perspective, in 1st person

# Tasks:
<details>
<summary>Click to expand</summary>

- [x] Lock the scheme to Lich King only
- [x] Cleared `block_plague_scheme`
- [x] Add the scheme game concept
- [x] Update the scheme game concept at the end
- [x] Confirm `plagued_rival_effect` works
- [x] Removed unwanted events from `plague_ongoing`
- [x] Adapt `plague_outcome.5002`
- [x] Fire `mx_cue_murder` only once, not in both event and on-action
- [x] Adapt `attempted_murder_opinion_effect`, add Plagued Me opinion
- [x] `plague_outcome.5001` tells when the Plague appears in your province instead of telling when the scheme fired
- [x] Check `wc_plague_effects`, they look half-finished
- [x] Check `wc_plague_scheme_discovery_events`, it may lack adaptation or be not-needed
- [x] Secrets don't work (or remove?)
  - [x] Adapt `secret_exposed_notification_effect`, it should send event
  - [x] Adapt `secret_exposed_owner_effects_effect`, it should add opinion
- [x] Move scheme changes from the Stun branch
- [x] `spread_plague_barony` may stop with the character death
- [x] Fix these:
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/138870856-8489d56a-cd2c-448c-9f79-4f03a31ad198.png)

</details>

- [x] Adapt this:
  - [x] Ner'zhul shouldn't be able to draw a map himself
  - [x] You don't need to draw a map of Terenas' castle since the scheme targets different barony
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/139255434-1f29d730-e274-41b6-8187-8f1b628898f4.png)

</details>

- [x] Revert test changes in the plague and murder schemes
- [x] Those who joined the Plague scheme convert to Lich King's religion once the scheme is over
- [x] Stop Plague at some point
- [x] Add Plague feature to Lich King trait
- [x] Make `on_invalidated`
- [x] Scourge declare war
- [x] Check `scheme` error
- [x] Owner to reveal not showing
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/140610727-c4fb5148-f7ed-438e-af84-c6199a346a4b.png)

</details>

- [x] Ner'zhul followers shouldn't have negative reaction to Plague
- [x] Add scheme cooldown in 603
- [x] Plague scheme has more agents but they give less progress
- [x] The Scourge not getting a notification on historical Plague
- [x] Make Plague work without Plaguer
- [x] Create Plague CB
</details>

# Tests:
- [x] Make sure everything works when the scheme is discovered

## How to test:
Start Ner'zhul in the year 603, right-click yourself, Debug Interactions, Give Dark Frenzy a few times so you have enough to start the Plague.

Right-click your target, Start a Plague.

`event WCHEA.1001` to get the Plague of Undeath.

To see the Plague spreading open either the Culture or Faith maps, Plagued provinces will have the Scourge culture and Mord'truari faith.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/647/files?authenticity_token=yd8G%2FHS%2BmXG01lp9JVYPet8MR4u4q9vH%2FuySfJKS%2BpPy9cyohgddB%2Bac4erFGpM7ffICnwv2br%2BBEGzOAXkKMg%3D%3D&file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here's the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).
